### ISSUE TITLE
Proposal: Config for offline usage

### ISSUE THREAD

**Issue by esparta** [indirect/unpwn#7]
Intro
===

The default usage of Unpwn assume once password to challenge pass the local review of common passwords (via bloom database) it should always consult [Have I Been Pwned][hibp](HIBP) vía [pwned gem][pwned], but they may be use case when there's no need to go in that route:

- Test suites
- Offline development

Proposal
===

I'm proposing to have the minimal config to avoid the usage of the HIBP on per request basis:

```ruby
def setup
  Unpwn.offline!
end
#... do the testing
def teardown
  Unpwn.online!
end
```

Rationale
===

Is generally accepted the _best practice_ of not use internet service directly or indirectly to test a module or function, not only because it will definitely will be slower than it requires, but also because in case of free services may have acceptable usage policies. In this case, it may happen because the users of Unpwn may, or may not be aware of this consequences when used at scale - as 50 test suites running in parallel and hitting HIBP a thousand times in a second.

Conclusions
===

Most people will be able to mock or monkey-patch the internals of Unpwn to have the same results, but it would be nice to have a way to just not hit internet at certain cases as testing or offline development.

If accepted, I do volunteer myself to implement the most simple approach to get this done.

[hibp]: https://haveibeenpwned.com/
[pwned]: https://rubygems.org/gems/pwned

**Comment by indirect** [comment]
That sounds great! Thank you for writing such a clear proposal. :)
